### ISSUE TITLE
Maximum Stack Size Exceeded error on query

### ISSUE THREAD

**Issue by thormuller** [AlecAivazis/houdini#141]
I started getting the following error on one particular query (running v0.9.8). The query runs successfully, but Houdini isn't happy. How should I understand this error, and any tips for resolving?

```
Maximum call stack size exceeded

RangeError: Maximum call stack size exceeded
    at Record.forgetSubscribers_walk (/$houdini/runtime/cache/record.js:107:27)
    at Record.forgetSubscribers_walk (/$houdini/runtime/cache/record.js:123:106)
    at Record.forgetSubscribers_walk (/$houdini/runtime/cache/record.js:123:106)
    at Record.forgetSubscribers_walk (/$houdini/runtime/cache/record.js:123:106)
    at Record.forgetSubscribers_walk (/$houdini/runtime/cache/record.js:123:106)
    at Record.forgetSubscribers_walk (/$houdini/runtime/cache/record.js:123:106)
    at Record.forgetSubscribers_walk (/$houdini/runtime/cache/record.js:123:106)
    at Record.forgetSubscribers_walk (/$houdini/runtime/cache/record.js:123:106)
    at Record.forgetSubscribers_walk (/$houdini/runtime/cache/record.js:123:106)
    at Record.forgetSubscribers_walk (/$houdini/runtime/cache/record.js:123:106)
```

The query looks like this:
```
const { data } = query<MyRecipeVersion>(graphql`
		query MyRecipeVersion($id: ID!, $offset: Int!) {
			getRecipe(id: $id) {
				id
				title
				versions(first: 1, offset: $offset) {
					ancestors {
						category
						completed
						created
					}
					id
					description
				}
			}
		}
	`);
```

**Comment by AlecAivazis** [comment]
So this error is caused when houdini notices that a record that was previously found in `ancestors` is no longer found and goes to unsubscribe the query's store from the result. It's almost certainly a bug. Do you mind sharing the artifact contents for the query?

**Comment by thormuller** [comment]
Thanks to your clear explanation I was able to debug the root cause of this error (indeed there was a hidden recursion in the query). This was not a Houdini bug. Thanks, Alec!

**Comment by AlecAivazis** [comment]
I'd like to make sure that the cache doesn't explode like this. a recursive reference isn't a bug in some data models. Do you mind sharing your query body?

**Comment by thormuller** [comment]
Here's the query response that breaks the cache:

```
{"getRecipe":{"id":"0x255685ffe","myTitle":"Frankie Recipe ","versions":[{"ingredients":null,"instructions":null,"keywords":null,"origin":null,"prepTime":null,"totalTime":null,"versionTitle":null,"yield":null,"image":null,"id":"0x255687234","diet":null,"description":null,"cuisine":null,"created":1626856166285,"cookTime":null,"completed":1626927473269,"category":null,"ancestors":[{"category":null,"completed":1626833547740,"cookTime":null,"created":1626822956349,"cuisine":null,"id":"0x255686008","descendents":[{"id":"0x255687234","versionTitle":null,"created":1626856166285}]}]}]}}
```

and here's the full query:

```
	query MyRecipeVersion($id: ID!, $offset: Int!) {
			getRecipe(id: $id) {
				id
				myTitle
				versions(first: 1, offset: $offset) {
					ingredients
					instructions
					keywords
					origin
					prepTime
					totalTime
					versionTitle
					yield
					image
					id
					diet
					description
					cuisine
					created
					cookTime
					completed
					category
					ancestors {
						category
						completed
						cookTime
						created
						cuisine
						id
						descendents {
							id
							versionTitle
							created
						}
					}
				}
			}
		}
```

**Comment by AlecAivazis** [comment]
fantastic. Thanks for sharing! Hope you're still enjoying working with houdini :)

**Comment by AlecAivazis** [comment]
This should now be fixed in `0.9.8` - do you mind reintroducing the bug on your end and making sure nothing blows up?

**Comment by thormuller** [comment]
Confirmed. This works with no errors given the previous query.

And to answer your earlier question--yes, loving Houdini now that it's well integrated into my workflow. Getting better each week, too. Looking forward to putting it into production one day soon.
